src/core/lib/slice/slice_internal.h

```diff
@@ -240,6 +242,19 @@ void grpc_slice_buffer_partial_unref_internal(grpc_slice_buffer* sb,
                                               size_t idx);
 void grpc_slice_buffer_destroy_internal(grpc_slice_buffer* sb);
 
+// Returns the first slice in the slice buffer.
+inline grpc_slice* grpc_slice_buffer_peek_first(grpc_slice_buffer* sb) {
+  GPR_DEBUG_ASSERT(sb->count > 0);
+  return &sb->slices[0];
+}
+
+// Consumes the first slice in the slice buffer.
+void grpc_slice_buffer_consume_first(grpc_slice_buffer* sb);
```

Comment:
So thinking of producer-consumer synchronization and other ways in which the word "consume" is commonly used, I would expect this to make that slice usable based on the name. So perhaps there's a different name choice out there like remove?

Reply:
I too like `remove` better. Changed.